### PR TITLE
add utility to compute self time of events

### PR DESCRIPTION
**Pull request by davidchencsl**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #79631

Purpose:
Compute self time for the experimental_event_tree
Test Plan:
Added test_utils_compute_self_time in test_profiler.py that verifies the computation.

### CODE REVIEW

File: torch/profiler/utils.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from collections import deque
```
Comment (robieta) [inline torch/profiler/utils.py:1]: We should name this `_utils.py` to avoid introducing symbols to the public namespace.

**Review by robieta — APPROVED** [review]
I had a single minor nit, but overall looks great!

File: torch/profiler/utils.py (hunk 1 above)
Comment (davidchencsl) [reply torch/profiler/utils.py:1]: Resolved

**Pull request closed without merging**